src/server/streamableHttp.ts

```diff
@@ -379,8 +451,25 @@ export class StreamableHTTPServerTransport implements Transport {
       // If the message is a response, use the request ID from the message
       requestId = message.id;
     }
+
+    // Check if this message should be sent on the standalone SSE stream (no request ID)
+    // Ignore notifications from tools (which have relatedRequestId set)
+    // Those will be sent via dedicated response SSE streams
     if (requestId === undefined) {
-      throw new Error("No request ID provided for the message");
+      // For standalone SSE streams, we can only send requests and notifications
+      if ('result' in message || 'error' in message) {
+        throw new Error("Cannot send a response on a standalone SSE stream unless resuming a previous client request");
+      }
+
+      const standaloneStream = this._responseMapping.get(this._sseStreamKey);
+      if (standaloneStream === undefined) {
+        // The spec says the server MAY send messages on the stream, so it's ok to discard if no stream
```

Comment:
Possible follow-up: do we want to offer an API that lets the implementor enqueue them for later delivery? (e.g., if the client is not connected now but might be later)